tests/core/tests/test_widgets.py

```diff
@@ -280,19 +282,31 @@ def test_clean_empty_string(self):
         self.assertEqual(self.widget.clean(" "), None)
         self.assertEqual(self.widget.clean("\r\n\t"), None)
 
+    @skipUnless(django.VERSION[0] < 4, f"skipping django {django.VERSION} version specific test")
+    @override_settings(LANGUAGE_CODE='fr-fr', USE_L10N=True)
+    def test_locale_render_lt_django4(self):
+        self.assertEqual("11,111", self.widget_coerce_to_string.render(self.value))
+
+    @skipUnless(django.VERSION[0] >= 4, f"skipping django {django.VERSION} version specific test")
+    @override_settings(LANGUAGE_CODE='fr-fr')
+    def test_locale_render_gte_django4(self):
+        self.assertEqual("11,111", self.widget_coerce_to_string.render(self.value))
+
 
 class DecimalWidgetTest(TestCase):
```

Comment:
I'm not sure to understand, you want to create a `test_locale_render` for each class ? Because Decimal and Float already have one and I don't see any TestCase for NumberWidget (I can create one but seems a bit redundant).

Reply:
Personally I would create a test case for each.  This is to protect against someone changing the implementation in a year's time and inadvertently breaking it.  I would recommend we have a test for the correct value being returned for each in the case where `coerce_to_string` is True and when it is False.

I don't think you need to have a test for `test_locale_render_lt_django4()`.  The result would be the same regardless of the django version if `use_l10n` is set.

Reply:
I will write a test case for each then. Regarding the Django version, I created two tests because starting from version 4, `USE_L10N` is set to `True` by default and I wanted to explicit the difference. Maybe a note about it in the docstring would be better.